occupancy_model/parse_footfall_pic_ble_logs.py

```
@@ -65,15 +63,32 @@ def conditional_formatting(save_path, sheet_name):
     red_fill = PatternFill(bgColor="FFC7CE")
     green_fill = PatternFill(bgColor="00CCFFCC")
     report_length = len(report["K"])
+    # Check pass condidion
     report.conditional_formatting.add(f'K1:K{report_length}', formatting.rule.CellIsRule(
         operator='lessThan', formula=['97'], fill=red_fill))
     report.conditional_formatting.add(f'K1:K{report_length}', formatting.rule.CellIsRule(
         operator='greaterThanOrEqual', formula=['97'], fill=green_fill))
+    # Add average accuracy
     report[f'K{report_length+1}'] = f'=AVERAGE(K1:K{report_length}'
     report[f'J{report_length+1}'] = f'=AVERAGE(J1:J{report_length}'
     report[f'I{report_length+1}'] = f'=AVERAGE(I1:I{report_length}'
+    # Fix column width at optimal
+    columns_widen(report)
     wb.save(save_path)
 
+def columns_widen(worksheet):
+    for col in worksheet.columns:
+     max_length = 0
+     column = col[0].column_letter
+     for cell in col:
+         try:
+             if len(str(cell.value)) > max_length:
+                 max_length = len(str(cell.value))
```

Maybe replace try/except block with:
max_length = max(len(str(cell.value)), max_length)?
or even better the whole loop with generator
max_length = max(len(str(cell.value)) for cell in col)

---

Btw using bare except is considered a bad practice, you should always specify which exceptions you want to capture, another way you may silence too much.

---

Have you checked that it works before requesting?